File: samples/CS/17-QuickAccessToolbar/RibbonItems.cs
using System;
using System.IO;
using System.Windows.Forms;


namespace WinForms.Ribbon
{
    partial class RibbonItems
    {
        private Stream _stream;

        public void Init()
        {
            ButtonNew.Click += new EventHandler<EventArgs>(_buttonNew_ExecuteEvent);
            ButtonSave.Click += new EventHandler<EventArgs>(_buttonSave_ExecuteEvent);
            ButtonOpen.Click += new EventHandler<EventArgs>(_buttonOpen_ExecuteEvent);

            // register to the QAT customize button
            QAT.Click += new EventHandler<EventArgs>(_ribbonQuickAccessToolbar_ExecuteEvent);
        }

        void _buttonNew_ExecuteEvent(object sender, EventArgs e)
        {
            // changing QAT commands list 
            UICollection<QatCommandPropertySet> itemsSource = QAT.QatItemsSource;
            itemsSource.Clear();
            itemsSource.Add(new QatCommandPropertySet() { CommandId = (uint)Cmd.cmdButtonNew });
            itemsSource.Add(new QatCommandPropertySet() { CommandId = (uint)Cmd.cmdButtonOpen });
            itemsSource.Add(new QatCommandPropertySet() { CommandId = (uint)Cmd.cmdButtonSave });
        }

        void _buttonSave_ExecuteEvent(object sender, EventArgs e)
        {
            // save ribbon QAT settings 
            _stream = new MemoryStream();
            Ribbon.SaveSettingsToStream(_stream);
        }

        void _buttonOpen_ExecuteEvent(object sender, EventArgs e)
        {
            if (_stream == null)
            {
                return;
            }

            // load ribbon QAT settings 
            _stream.Position = 0;
            Ribbon.LoadSettingsFromStream(_stream);
        }

        void _ribbonQuickAccessToolbar_ExecuteEvent(object sender, EventArgs e)
        {
            MessageBox.Show("Open customize commands dialog..");
        }

        public void Load()
        {
        }

    }
}

File: samples/VB/17-QuickAccessToolbar/RibbonItems.vb
Imports System
Imports System.IO

Namespace WinForms.Ribbon

    Partial Class RibbonItems
		Private _stream As Stream

		Public Sub Init()
			AddHandler _ButtonNew.Click, AddressOf _buttonNew_ExecuteEvent
			AddHandler _ButtonSave.Click, AddressOf _buttonSave_ExecuteEvent
			AddHandler _ButtonOpen.Click, AddressOf _buttonOpen_ExecuteEvent

			' register to the QAT customize button
			AddHandler QAT.Click, AddressOf _ribbonQuickAccessToolbar_ExecuteEvent
		End Sub

		Private Sub _buttonNew_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			' changing QAT commands list 
			Dim itemsSource As UICollection(Of QatCommandPropertySet) = QAT.QatItemsSource
			itemsSource.Clear()
			itemsSource.Add(New QatCommandPropertySet() With {.CommandId = CUInt(Cmd.cmdButtonNew)})
			itemsSource.Add(New QatCommandPropertySet() With {.CommandId = CUInt(Cmd.cmdButtonOpen)})
			itemsSource.Add(New QatCommandPropertySet() With {.CommandId = CUInt(Cmd.cmdButtonSave)})
		End Sub

		Private Sub _buttonSave_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			' save ribbon QAT settings 
			_stream = New MemoryStream()
			_ribbon.SaveSettingsToStream(_stream)
		End Sub

		Private Sub _buttonOpen_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			If _stream Is Nothing Then
				Return
			End If

			' load ribbon QAT settings 
			_stream.Position = 0
			_ribbon.LoadSettingsFromStream(_stream)
		End Sub

		Private Sub _ribbonQuickAccessToolbar_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			MessageBox.Show("Open customize commands dialog..")
		End Sub

	End Class
End Namespace
